File: swift/numbers/Sources/numbers_recursive/numbers_recursive.swift
public struct NumbersRecursive {
    public static func fibonacci(_ n: Int) -> Int {
        if n <= 1 { return n }
        return fibonacci(n - 1) + fibonacci(n - 2)
    }
    
    public static func factorial(_ n: Int) -> Int {
        if n <= 1 { return 1 }
        return n * factorial(n - 1)
    }
    
    public static func sumNumbers(_ n: Int) -> Int {
        if n == 0 { return 0 }
        return n + sumNumbers(n - 1)
    }
}

File: _experimental/_scala/numbers/src/test/scala/numbers_recursive.scala
import org.scalatest.funsuite.AnyFunSuite

class NumbersRecursiveTest extends AnyFunSuite {
  val nr = new NumbersRecursive

  test("Fibonacci Recursive") {
    assert(nr.fibonacci(5) == 5)
  }

  test("Factorial Recursive") {
    assert(nr.factorial(5) == 120)
  }

  test("Sum_Numbers Recursive") {
    assert(nr.sum_numbers(5) == 15)
  }
}